src/components/errorbars/plot.js

```diff
@@ -77,11 +111,26 @@ module.exports = function plot(traces, plotinfo) {
 
                 if(!coords.noXS) path += 'm0,-' + xw + 'v' + (2 * xw); // shoe
 
-                errorbar.append('path')
-                    .classed('xerror', true)
-                    .attr('d', path);
+                var xerror = errorbar.select('path.xerror');
+
+                isNew = !xerror.size();
+
+                if(isNew) {
+                    xerror = errorbar.append('path')
+                        .classed('xerror', true);
+                } else if(hasAnimation) {
+                    xerror = xerror
+                        .transition()
+                            .duration(transitionConfig.duration)
+                            .ease(transitionConfig.ease)
+                            .delay(transitionConfig.delay);
+                }
+
+                xerror.attr('d', path);
             }
         });
+
+        d3.select(this).call(styleError);
```

Comment:
We should only call `styleError` on animations.

Otherwise, it would get call twice as it is already part of [`Plots.style`](https://github.com/plotly/plotly.js/blob/5b92305206ec84df05fddda65b7f4033170b3bb7/src/plots/plots.js#L825-L833)